v2/api/managedidentity/customizations/user_assigned_identity_extention_authorization.go

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package customizations
+
+import (
+	"context"
+	"fmt"
+
+	v20230131s "github.com/Azure/azure-service-operator/v2/api/managedidentity/v1api20230131/storage"
+	"github.com/Azure/azure-service-operator/v2/internal/genericarmclient"
+	. "github.com/Azure/azure-service-operator/v2/internal/logging"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/secrets"
+	"github.com/go-logr/logr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+)
+
+var _ genruntime.KubernetesExporter = &UserAssignedIdentityExtension{}
+
+// ExportKubernetesResources defines a resource which can create other resources in Kubernetes.
+func (identity *UserAssignedIdentityExtension) ExportKubernetesResources(
+	ctx context.Context,
+	obj genruntime.MetaObject,
+	armClient *genericarmclient.GenericClient,
+	log logr.Logger,
+) ([]client.Object, error) {
+	typedObj, ok := obj.(*v20230131s.UserAssignedIdentity)
+	if !ok {
+		return nil, fmt.Errorf(
+			"cannot run on unknown resource type %T, expected *v20230131s.UserAssignedIdentity", obj)
+	}
+
+	// Type assert that we are the hub type. This will fail to compile if
+	// the hub type has been changed but this extension has not
+	var _ conversion.Hub = typedObj
+
+	hasSecrets := secretsSpecified(typedObj)
+	if !hasSecrets {
+		log.V(Debug).Info("No secrets retrieval to perform as operatorSpec.Secrets is empty")
+		return nil, nil
+	}
+
+	// obj.Status.ClientId
+
+	collector := secrets.NewCollector(typedObj.Namespace)
+	if typedObj.Spec.OperatorSpec != nil && typedObj.Spec.OperatorSpec.Secrets != nil {
```

Comment:
This can be rewritten like so:

```
if typedObj.Spec.OperatorSpec != nil && typedObj.Spec.OperatorSpec.Secrets != nil {
    collector.AddValue(typedObj.Spec.OperatorSpec.Secrets.ClientId, to.Value(typedObj.Status.ClientId))
    collector.AddValue(typedObj.Spec.OperatorSpec.Secrets.PrincipalId, t.Value(typedObj.Status.PrincipalId))
    collector.AddValue(typedObj.Spec.OperatorSpec.Secrets.TenantId, t.Value(typedObj.Status.TenantId))
}
```

The `AddValue` function handles the required nil-checking for the left and right parameters.

Reply:
Refactored.